File: base/go/rwfiles.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"math/rand"
	"os"
	"strconv"
)

func compute() {

	writes, _ := strconv.Atoi(os.Getenv("WRITES"))
	reads, _ := strconv.Atoi(os.Getenv("READS"))
	fileSize, _ := strconv.Atoi(os.Getenv("FILE_SIZE"))

	//create the data array
	data := make([]byte, fileSize)

	// write the files
	for i := 0; i < writes; i++ {
		path := fmt.Sprintf("/data/%v", i)
		err := ioutil.WriteFile(path, data, 0644)
		if err != nil {
			log.Fatal(err)
		}
	}

	//read the files
	for i := 0; i < reads; i++ {
		path := fmt.Sprintf("/data/%v", rand.Intn(writes))
		_, err := ioutil.ReadFile(path)
		if err != nil {
			panic(err)
		}
	}

}

File: base/python/rwfiles.py
import os
import time
from random import seed
from random import randint

# seed random number generator
seed(time.time())

def compute():

    writes = int(os.environ['WRITES'])
    reads = int(os.environ['READS'])
    file_size = int(os.environ['FILE_SIZE'])

    # create an array with the file names
    files = [""] * writes

    # create the file data
    data = "0" * file_size

    # write the files
    for i in range(writes):
        path = "/data/{}".format(i)
        f = open(path, "w")
        f.write(data)
        f.close()

    # read the files
    for i in range(reads):
        path = "/data/{}".format(randint(0, writes-1))
        f = open(path, "r")
        f.read()
